src/platform/Kconfig

```diff
@@ -490,6 +490,7 @@ config SYSTICK_PERIOD
 
 config HAVE_AGENT
 	bool "Enable system agent"
+	default n if ZEPHYR_SOF_MODULE
 	default y
 	help
```

Comment:
My only concern here is how does this work if ZEPHYR_SOF_MODULE=y

We will have:

``config HAVE_AGENT
       bool "xxx"
       default n
       default y
``

Reply:
It takes the first thing seen:

> default value: "default" ["if" ] A config option can have any number of default values. If multiple default values are visible, only the first defined one is active.

Source: https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt